### PR TITLE
Breaking change: php 7.2 compatibility

### PR DESCRIPTION
**Pull request by prevostc**
I could not find a way to keep php 5 compat.
Works with php 7.2, maybe 7.1.

### CODE REVIEW

**Comment by prevostc** [comment]
I did downgrade symfony/process to 3.4.4. This allows php 5.6 users to transition smoothly.

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 /vendor/
 composer.lock
+.idea
```
Comment (krichprollsch) [inline .gitignore]: Please don't add your ide's specific files here, configure a global `.gitignore` to do that.
https://help.github.com/articles/ignoring-files/#create-a-global-gitignore

**Review by krichprollsch — CHANGES_REQUESTED** [review]

**Comment by krichprollsch** [comment]
That's very nice ! thank you very much 🌮

I'm not sure we must keep the back compat with old versions of php (even with `7.0`).
Composer is able to ignore non-compatible php version requirement.
So we can set the composer requirement to 7.1+ and keep the deps with sf components `4.x`.

WDYT @prevostc ?

**Comment by krichprollsch** [comment]
That's very nice ! thank you very much 🌮

I'm not sure we must keep the back compat with old versions of php (even with `7.0`).
Composer is able to ignore non-compatible php version requirement.
So we can set the composer requirement to 7.1+ and keep the deps with sf components `4.x`.

WDYT @prevostc ?

File: .gitignore (hunk 1 above)
Comment (prevostc) [reply .gitignore]: Good advice, I did not know about this 👍

**Comment by prevostc** [comment]
I personally still need the php5.6 compatibility for some time. As I slowly migrate non-php7-compatible code to php7-compatible code in dev, I will still use php5 in production.

The php-dmtx code is basically the same when we use symfony/process v3.6 and v4, I suggest we use the semver `"symfony/process": "~3.4||~4.0"` and php requirements to `">=5.6"` to content everyone. php7+ will use `symfony/process` v4 and php5.6 will use v3.4.

**Comment by krichprollsch** [comment]
I'm ok for that. Could you just squash your commits please? And I merge 🚀

**Comment by prevostc** [comment]
I'll do that on Monday, thx.

**Comment by prevostc** [comment]
Cleaned the .gitignore and squashed all commits

**Pull request merged**